### PR TITLE
MC particles table: fill status codes and fix typo in flags

### PR DESCRIPTION
**Pull request by nburmaso**
Following PR #7483

### CODE REVIEW

**Review by sawenzel — DISMISSED** [review]

**Comment by sawenzel** [comment]
@amorsch : This is what we discussed. @jgrosseo : I think it would make sense to offer a (dynamic columns) API to allow extracting physics processes for secondaries (via the status code storage). If the particle is in fact a primary ... the physics code should be set to TMCProcess::kPrimary

**Comment by jgrosseo** [comment]
I am confused now: Is the "from transport" bit now used for isPrimary?

Sure, we can add a dynamic column. If you give me the exact mapping, I'll add it.

**Comment by sawenzel** [comment]
@jgrosseo : I think there is a "not" operator in front. So not primary is from transport.

**Comment by sawenzel** [comment]
See also https://github.com/AliceO2Group/AliceO2/pull/7492 in relation to this.

File: Detectors/AOD/src/AODProducerWorkflowSpec.cxx [hunk 1]
```diff
@@ -658,8 +658,11 @@ void AODProducerWorkflowDPL::fillMCParticlesTable(o2::steer::MCKinematicsReader&
       }
       int statusCode = 0;
       uint8_t flags = 0;
-      if (!mcParticles[particle].isTransported()) {
+      if (!mcParticles[particle].isPrimary()) {
         flags |= 1 << 0; // mark as transported
```
Comment (sawenzel) [inline Detectors/AOD/src/AODProducerWorkflowSpec.cxx]: the comment is misleading. I would say `// mark as produced by transport`

Comment (jgrosseo) [reply Detectors/AOD/src/AODProducerWorkflowSpec.cxx]: Yes, and also use o2::aod::mcparticle::enums::ProducedByTransport instead of `1 << 0`

Comment (nburmaso) [reply Detectors/AOD/src/AODProducerWorkflowSpec.cxx]: Thank you for comments. I have cleaned up the code

**Pull request merged**